DFC.ServiceTaxonomy.GraphVisualiser/Controllers/VisualiseController.cs

```diff
@@ -91,18 +148,54 @@ private ActionResult GetOntology()
             return Content(owlResponseString, MediaTypeNames.Application.Json);
         }
 
-        private async Task<ActionResult> GetData(string uri, string graph)
+        private async Task<ActionResult> GetData(string contentItemId, string graph)
         {
-            const string prefLabel = "skos__prefLabel";
-            var query = new GetNodesCypherQuery(nameof(uri), uri, prefLabel, prefLabel);
+            IEnumerable<string> relationshipCommands = await BuildCommands(contentItemId);
 
-            //todo: allow user to visualise published and draft databases. new story?
-            await _neoGraphCluster.Run(graph, query);
+            var resultList = new List<INodeAndOutRelationshipsAndTheirInRelationships?>();
 
-            var owlDataModel = _neo4JToOwlGeneratorService.CreateOwlDataModels(query.SelectedNodeId, query.Nodes, query.Relationships, prefLabel);
-            var owlResponseString = JsonSerializer.Serialize(owlDataModel, _jsonOptions);
+            foreach (var command in relationshipCommands)
+            {
+                var result = await _neoGraphCluster.Run(graph, new NodeAndOutRelationshipsAndTheirInRelationshipsQueryFromCommand(command!));
+                resultList.AddRange(result);
+            }
+
+            string owlResponseString = "";
+
+            if (resultList.Any())
+            {
+                //Get all outgoing relationships from the query and add in any source nodes
+                var allNodeOutgoingRelationships = resultList.SelectMany(x => x!.OutgoingRelationships.Select(x => x.outgoingRelationship.DestinationNode)).Union(resultList.GroupBy(x => x!.SourceNode).Select(z => z.FirstOrDefault()!.SourceNode));
+
+                var owlDataModel = _neo4JToOwlGeneratorService.CreateOwlDataModels(resultList.FirstOrDefault()!.SourceNode.Id, allNodeOutgoingRelationships, resultList!.SelectMany(y => y!.OutgoingRelationships.Select(z => z.outgoingRelationship.Relationship)).ToHashSet<IRelationship>(), "skos__prefLabel");
+                owlResponseString = JsonSerializer.Serialize(owlDataModel, _jsonOptions);
+            }
+            else
+            {
+                owlResponseString = "{}";
+            }
 
             return Content(owlResponseString, MediaTypeNames.Application.Json);
         }
+
+        private async Task<IEnumerable<string>> BuildCommands(string contentItemId)
```

Comment:
This code should be split out into a syncer class, at the same location as e.g. `MergeGraphSyncer`.
It should then make it possible to call back into for embedded/picked items.